compiler/rustc_middle/src/ty/predicate.rs

```diff
@@ -137,6 +136,7 @@ impl<'tcx> Predicate<'tcx> {
             | PredicateKind::Coerce(_)
             | PredicateKind::Clause(ClauseKind::ConstEvaluatable(_))
             | PredicateKind::ConstEquate(_, _)
+            | PredicateKind::NormalizesTo(..)
             | PredicateKind::Ambiguous => true,
```

Comment:
we're actually free to normalize `NormalizesTo` goals. The alias we're normalizing is already an `AliasTerm` so we only ever normalize the arguments, not the alias itself.

We could in theory also normalize the arguments of aliases in `AliasRelate`. However, setting `allow_normalization` to `true` for this would cause us to also attempt to normalize the aliases we're actually relating. Given that we use `AliasRelate` to do so, that would result in a trivial cycle.